manifests/server/access.pp

```diff
@@ -1,23 +1,27 @@
 # See README.md for details.
 define openldap::server::access (
-  Optional[Enum['present', 'absent']]  $ensure   = undef,
-  Optional[Variant[Integer,String[1]]] $position = undef, # FIXME We should probably choose Integer or String
-  Optional[String[1]]                  $what     = undef,
-  Optional[String[1]]                  $suffix   = undef,
-  Optional[Array[String[1]]]           $access   = undef,
+  String[1]                            $what,
+  Array[Openldap::Access_rule]         $access,
+  Enum['present', 'absent']            $ensure   = 'present',
+  Optional[Variant[Integer,String[1]]] $position = undef, # FIXME deprecated
+  Optional[String[1]]                  $suffix   = undef, # FIXME deprecated
```

Comment:
Good catch. I was feeling bad because I am afraid of meeting this guy in the street:

![I am once again asking you stop breaking the API](https://user-images.githubusercontent.com/148721/136709998-2293acab-5294-49bc-afc7-c4aca42e84a9.png)

Reply:
The next release being a major one, I decided to remove the legacy code completely and added a commit for this.

Reply:
The commit has been squashed for the sake of simplicity :wink: